### PR TITLE
Add HB_NO_OEPN

### LINKED ISSUE #1764: Suggestions for HB_TINY additions
Would be nice if we had some definition for removing use of `setlocale` and disable `hb_blob_create_from_file` use of fopen/fclose/...

Also not sure why we have a round [polyfill](https://github.com/harfbuzz/harfbuzz/blob/3c240bd3dc0aaca38154da555d0aef350da62ee6/src/hb.hh#L518) but isn't used anywhere or instead roundf, probably should be deleted altogether.

### PR DESCRIPTION
**Pull request by ebraminio**
But don't see the need to add it to lean or tiny personally, what motivated me for PR is to ask about the naming

(fixes #1764)

### CODE REVIEW

**Comment by behdad** [comment]
Rename to HB_NO_OPEN?  We also have HB_NO_MMAP.   Let's add both to HB_TINY.

Also, I had a change of mind: I think in HB_TINY, or in general, API that gets disabled should be removed instead of returning empty/error/false.  That way we know if a client links it will work.  And if it doesn't link it will point to which feature needs to be re-enabled.  I'll go ahead and do that.

**Comment by ebraminio** [comment]
> Also, I had a change of mind: I think in HB_TINY, or in general, API that gets disabled should be removed instead of returning empty/error/false. That way we know if a client links it will work. And if it doesn't link it will point to which feature needs to be re-enabled. I'll go ahead and do that.

Yep, makes lots of sense, I just fear from compile warnings :/

**Comment by behdad** [comment]
> Yep, makes lots of sense, I just fear from compile warnings :/

Right :(.  Let me see what I can come up with.

File: src/hb-config.hh [hunk 1]
```diff
@@ -60,6 +61,7 @@
 #define HB_NO_GETENV
 #define HB_NO_LAYOUT_UNUSED
 #define HB_NO_MATH
+#define HB_NO_MMAP
```
Comment (ebraminio) [inline src/hb-config.hh]: Or having both in lean? My reasoning was it can work without mmap so it is lean but disabling the whole API in tiny

Comment (behdad) [reply src/hb-config.hh]: Right. TINY itself doesn't do these; LEAN does.  IMplement it by disabling the feature now.  Then we switch all to remove the API.  I don't think we need both.

**Pull request merged**